azavea/echo-locator
# Set up production Amplify stack

## Set up new Amplify stack for production

Configures new Amplify stack for production following the steps in the `staging` environment `README.md` for adding the counselor user group, using emails as usernames, and  adding the ability to create client user accounts.


### Notes

It was necessary to update the Amplify CLI version in use to generate the new stack (now on 4.13.1).

The files here were autogenerated by the Amplify CLI.

Once the new environment has been adopted, the old Amplify stack in the `production` environment directory should be destroyed using the Amplify CLI. Once the CLI has torn down the resources successfully, the existing `production` environment directory should be deleted and the new environment directory added here, `newprod`, should be renamed to `production`. I'll change this issue from `WIP` when that's done.


## Testing Instructions

 * Copy the configuration file to the app: `cp deployment/amplify/newprod/src/aws-exports.js taui/src/aws-exports.js`
 * Create a new user account in Cognito and add it to the `counselors` group
 * Site should still function as expected
 * Cognito emails should look as expected
 * Existing profiles copied over from the last production environment should be accessible
 * The AWS Athena query for the new profiles should work as expected and include the three new fields for user email, whether the email invite has been set, and whether the email has been confirmed by first sign-in


Resolves #58
